pkg/operator/observe_config.go

```diff
@@ -97,28 +106,29 @@ func (c ConfigObserver) sync() error {
 	var err error
 	observedConfig := map[string]interface{}{}
 
+	operatorConfig, err := c.operatorConfigClient.Get("instance", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	currentCASpec := operatorConfig.Spec.AdditionalTrustedCA.DeepCopy()
```

Comment:
Fetch operator config first, which can be mutated by the `observer` functions.
Deep copy the current CA spec so we can exit early if no updates are needed.